### PR TITLE
rpc: add error when call result parameter is not addressable

### LINKED ISSUE #20628: rpc: Complain when `Call` result parameter is not assignable
It's easy to make a typo like 
```
var result
client.rpc.Call(result, "admin_peers")
```
instead of pasing `&result` leads to silently not returning a result. It should generate an error.

### PR DESCRIPTION
**Pull request by adamschmideg**
Fixes #20628

### CODE REVIEW

File: rpc/client.go [hunk 1]
```diff
@@ -276,6 +276,9 @@ func (c *Client) Call(result interface{}, method string, args ...interface{}) er
 // The result must be a pointer so that package json can unmarshal into it. You
 // can also pass nil, in which case the result is ignored.
 func (c *Client) CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error {
+	if result != nil && reflect.TypeOf(result).Kind() != reflect.Ptr {
+		return fmt.Errorf("Expected nil or pointer for result, got %v", result)
```
Comment (fjl) [inline rpc/client.go]: Please change this error message to lowercase. I'd also phrase it slightly differently, maybe "call result parameter must be pointer or nil interface".

File: rpc/client_test.go [hunk 2]
```diff
@@ -49,6 +49,29 @@ func TestClientRequest(t *testing.T) {
...
+	server := newTestServer()
+	defer server.Stop()
+	client := DialInProc(server)
+	defer client.Close()
+
+	if err := client.Call(nil, "test_echo", "hello", 10, &echoArgs{"world"}); err != nil {
+		t.Errorf("Passing nil as result should be fine, but got an error: %v", err)
+	}
+	var resultVar echoResult
+	if err := client.Call(&resultVar, "test_echo", "hello", 10, &echoArgs{"world"}); err != nil {
```
Comment (fjl) [inline rpc/client_test.go]: You don't need to test this case because it is already covered by many other tests.

File: rpc/client.go (hunk 1 above)
Comment (adamschmideg) [reply rpc/client.go]: Done

File: rpc/client_test.go (hunk 2 above)
Comment (adamschmideg) [reply rpc/client_test.go]: Done

**Pull request merged**